File: Leetcode/cpp/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        /// we can't solve it easily by sorting due to we need return the index before sorting

        /// Use key as value of array
        /// Use val as index of array
        unordered_map<int,int> numToIdx;
        for(int i=0; i<nums.size(); i++) {
            if(numToIdx.count(target-nums[i])) {
                return {numToIdx[target-nums[i]], i};
            }
            /// Update key and value
            numToIdx[nums[i]] = i;
        }

        return {};
    }
};

File: Leetcode/java/1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> numToIdx = new HashMap<>();

        for(int i=0; i<nums.length; i++) {
            if(numToIdx.containsKey(target-nums[i])) {
                return new int [] { numToIdx.get(target-nums[i]), i};
            }
            numToIdx.put(nums[i], i);
        }

        return new int [] {};
    }
}